### PR TITLE
Feature-Reporting infrastructure-Issue-216

### PR DESCRIPTION
**Pull request by allan-allay**
In-app reporting library infrastructure for building indicator tallies, persisting & retrieving the tallies for display in a dashboard

### CODE REVIEW

**Comment by ndegwamartin** [comment]
@allan-allay could you fix out the codacy issues?

File: opensrp-reporting/build.gradle [hunk 1]
```diff
@@ -61,6 +61,7 @@ android {
     testOptions {
         unitTests {
             includeAndroidResources = true
+            returnDefaultValues = true
```
Comment (ndegwamartin) [inline opensrp-reporting/build.gradle]: Do we require this line? Might hide some regression bugs in future

File: sample/src/main/java/org/smartregister/sample/utils/SampleDataDBUtil.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    public static String TAG = SampleDataDBUtil.class.getCanonicalName();
+
+    public static void addSampleIndicators(IndicatorRepository indicatorRepository) {
+        indicatorRepository.add(new ReportIndicator(null, numericIndicatorKey, "Total Children", null));
+        indicatorRepository.add(new ReportIndicator(null, pieChartYesIndicatorKey, "Immunized children", null));
+        indicatorRepository.add(new ReportIndicator(null, pieChartNoIndicatorKey, "Non immunized children", null));
+    }
+
+    public static void addSampleIndicatorQueries(IndicatorQueryRepository indicatorQueryRepository) {
+        // IndicatorQueries with sample count queries
```
[the remark this answers is not in the document]
Comment (allan-allay) [reply sample/src/main/java/org/smartregister/sample/utils/SampleDataDBUtil.java]: Refactored adding/initialising indicator data + queries from a Yaml Config file. 
The config file will have the indicator properties and the application reads the file and persists the data in the app DB.

File: sample/src/main/java/org/smartregister/sample/utils/SampleDataDBUtil.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        indicatorQueryRepository.add(new IndicatorQuery(null, numericIndicatorKey, "select count(*) from " + IndicatorQueryRepository.INDICATOR_QUERY_TABLE, BuildConfig.DATABASE_VERSION));
+        indicatorQueryRepository.add(new IndicatorQuery(null, pieChartYesIndicatorKey, "select count(*) from " + IndicatorQueryRepository.INDICATOR_QUERY_TABLE, BuildConfig.DATABASE_VERSION));
```
[the remark this answers is not in the document]
Comment (allan-allay) [reply sample/src/main/java/org/smartregister/sample/utils/SampleDataDBUtil.java]: Refactored as advised above.

File: sample/src/main/java/org/smartregister/sample/utils/SampleDataDBUtil.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (allan-allay) [reply sample/src/main/java/org/smartregister/sample/utils/SampleDataDBUtil.java]: **See commits** 
[Commit](https://github.com/OpenSRP/opensrp-client-reporting/pull/3/commits/b957541fc138325970a23405abeb4f530d4c383b)

and 

[Commit](https://github.com/OpenSRP/opensrp-client-reporting/pull/3/commits/a7df827c2fabb3f3898f6f4e6606ee0024e1501b)

File: sample/src/main/java/org/smartregister/sample/utils/SampleDataDBUtil.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (allan-allay) [reply sample/src/main/java/org/smartregister/sample/utils/SampleDataDBUtil.java]: Refactored as adviced above: 

**See commits** 
[Commit](https://github.com/OpenSRP/opensrp-client-reporting/pull/3/commits/b957541fc138325970a23405abeb4f530d4c383b)

and 

[Commit](https://github.com/OpenSRP/opensrp-client-reporting/pull/3/commits/a7df827c2fabb3f3898f6f4e6606ee0024e1501b)

**Pull request merged**